### PR TITLE
Traitlets-based numprops

### PR DESCRIPTION
**Pull request by bluescarni**
@astrofrog 

Here's the pull request for the traitlets-based numprops. I have implemented the changes we discussed via email, updated the docs and bumped the version number. The behaviour should be identical to the old implementation.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+0.3 (unreleased)
+----------------
+
+- Implement on top of the traitlets module.
```
Comment (astrofrog) [inline CHANGES.md]: Actually you can add this to the 0.2 section

File: test_numprops.py [hunk 2]
```diff
@@ -311,26 +316,36 @@ def test_invalid_units(self):
...
 
 def test_invalid_unit_framework():
 
-    with pytest.raises(ValueError) as exc:
-        a = NumericalProperty('a', convertible_to='m')
+    with pytest.raises(TraitError) as exc:
+        a = NumericalProperty(convertible_to='m')
     assert exc.value.args[0] == "Could not identify unit framework for target unit of type str"
+
+def test_classes():
```
Comment (astrofrog) [inline test_numprops.py]: So does this run everything as a single test?

**Comment by astrofrog** [comment]
@bluescarni - thanks! Just a couple of comments above, and another one here: we should add a sentence or two at the start of the README to explain that this is built on top of traitlets, and explain what traitlets is.

**Comment by bluescarni** [comment]
@astrofrog I implemented the changes we discussed.

**Comment by astrofrog** [comment]
@bluescarni - sorry for the delay in merging this, and thanks for working on it!

**Pull request merged**